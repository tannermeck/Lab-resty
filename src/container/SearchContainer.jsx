import React, { Component } from 'react';
import { SearchForm } from '../presentation/SearchForm';

class SearchContainer extends Component {
    state = {
        urlSearch: '',
        method: '',
        jsonBody: {}
    }

    // handleSubmit = async () => {
    // }
    handleSearch = ({ target }) => {
        this.setState({urlSearch: target.value })
    }

    render() {
        console.log('search', this.state.urlSearch)
        return (
            <>
                <h1>RESTless</h1>
                <SearchForm search={this.handleSearch} />
            </>
        )
    }
}
export default SearchContainer;