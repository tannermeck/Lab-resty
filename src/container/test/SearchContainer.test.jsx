import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen, waitFor } from '@testing-library/react';
import SearchContainer from '../SearchContainer';

describe('it should display a todo', () => {
    it('should list one todo by its id', async () => {
        render(<SearchContainer />)
        const urlInput = await screen.findByPlaceholderText('URL')
        userEvent.type(urlInput, 'https://jsonplaceholder.typicode.com/todos/1')
        const submit = await screen.findByRole('button');
        userEvent.click(submit);
        return waitFor(() => {
            const data = screen.getAllByText('root', {exact: false})
            expect(data).toHaveLength(1);
        })
        // const ul = await screen.findByRole('list')
        // expect(ul).toMatchSnapshot();
    })
})