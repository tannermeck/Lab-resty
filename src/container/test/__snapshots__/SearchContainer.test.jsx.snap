// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should display a todo should list one todo by its id 1`] = `
<ul>
  <li>
    <p>
      GET
    </p>
    <p>
      https://jsonplaceholder.typicode.com/todos/1
    </p>
  </li>
</ul>
`;
